src/components/SubscriptionCard.js

```diff
@@ -366,38 +420,53 @@ class SubscriptionCard extends React.Component {
-                <span className='currency-frequency'> {subscription.currency.toLowerCase()}/{frequency} </span> 
-              </div>
+              {this.state.visibleState === this.stateConstants.editAmount &&
+                <div className="editAmount">
+                  <div className="inputAmount">
+                    <span className="currency">$</span>
+                    <input type="number" step="0.01" min="1" name="amount" value={this.state.amountValue || subscription.totalAmount / 100} onChange={this.handleAmountChange}/>
+                    <span className="frequency">{subscription.interval === 'year' ? 'Yearly' : 'Monthly'}</span>
+                  </div>
+                  <div className="update-buttons">
+                    <SmallButton className="no" bsStyle="primary" onClick={this.resetState}><FormattedMessage id="subscription.updateAmount.cancel.btn" defaultMessage="cancel" /></SmallButton>
+                    <SmallButton className="yes" bsStyle="primary" onClick={this.updateAmount}><FormattedMessage id="subscription.updateAmount.update.btn" defaultMessage="update" /></SmallButton>
+                  </div>
+                </div>
+              }
+              {this.state.visibleState !== this.stateConstants.editAmount &&
+                <div className={`amount-frequency ${subscription.isPastDue ? 'past-due' : ''}`}>
+                  <span className='amount'><Currency value={subscription.totalAmount} currency={subscription.currency} /></span>
+                  <span className='currency-frequency'> {subscription.currency.toLowerCase()}/{frequency} </span>
+                </div>
+              }
             </div>
 
-            {this.state.visibleState === 'normal' && subscription.stats && Object.keys(subscription.stats).length > 0 && 
-                  <div className='contribution'>
-                    <span className="totalAmount">
-                      <Currency value={get(subscription, 'stats.totalDonations')} currency={subscription.currency} />&nbsp;
-                    </span>
-                    <span className="total-amount-text">{intl.formatMessage(this.messages['subscription.amountToDate'])}</span>
-                  </div>}
-
+            {this.state.visibleState === 'normal' && subscription.stats && Object.keys(subscription.stats).length > 0 &&
+              <div className='contribution'>
+                <span className="totalAmount">
+                  <Currency value={get(subscription, 'stats.totalDonations')} currency={subscription.currency} />&nbsp;
+                </span>
+                <span className="total-amount-text">{intl.formatMessage(this.messages['subscription.amountToDate'])}</span>
+              </div>}
 
-            {this.state.visibleState !== this.stateConstants.cancelConf && subscription.paymentMethod && canEditSubscription && 
+            {this.state.visibleState !== this.stateConstants.cancelConf && subscription.paymentMethod && canEditSubscription &&
```

Comment:
I noticed that when you select "Update payment amount", whole subscription box gets bigger. It's fine when you only have one subscription. If you have a bunch of them in a row, all of them get bigger and create empty space at the bottom. 

Is it possible to remove payment method in `editAmount` view? Ideally, we could prevent box from growing at all ... or growing very little.

Reply:
I'll have a look.